### ISSUE TITLE
Set up GitHub Pages on flutter/plugins for integration tests

### ISSUE THREAD

**Issue by stuartmorgan** [flutter/flutter#95420]
We have integration tests for at least webview_flutter and video_player that are currently hitting external URLs, which is random test failure waiting to happen (this has already been a problem at least once). In the long term we probably need to have the tests run their own local server for some use cases, but for most of them just having a static site that:
- we can change in-repo,
- we know won't suddenly change out from under us (e.g., removal of a video we are pointing at), and
- won't have unexpected behaviors (like when a Google page we were pointing to started redirecting our tests to a "suspicious activity" page when run from CI)

would solve a lot of potential problems. We should be able to do that via GitHub pages running on flutter/plugins.

**Comment by stuartmorgan** [comment]
Retitling since Hixie demonstrated in https://github.com/flutter/flutter/issues/95420 that the local server approach is vastly easier than I expected.

**Comment by stuartmorgan** [comment]
What's currently left here:
- webview_flutter tests that try to load `youtube.com` while that's blocked by policy, and verify that it doesn't load. We *should* be able to change that to a local webserver URL, but when I tried that last I saw a weird issue where it seemed to be using the policy of the actual example app's controller (which blocks youtube.com) rather than the integration test. Ideally we should fix that, but it's relatively low priority given that it's not supposed to load anyway.
- some video_player tests that are using github asset URLs. Serving the data content correctly was not straightforward when I tried. We should really fix that, since it introduces a network dependency, but it's at least not a URL that's going to change behavior out from under us.

Downgrading to P2 to reflect the severity of what's left.
